src/lumigo_tracer/wrappers/aiohttp/aiohttp_wrapper.py

```diff
@@ -0,0 +1,87 @@
+
+try:
+    import aiohttp
+except Exception:
+    aiohttp = None
+
+LUMIGO_SPAN_ID_KEY = "_lumigo_span_id"
+trace_config = None
+
+
+def aiohttp_session_init_wrapper(func, instance, args, kwargs):
+    with lumigo_safe_execute("aiohttp aiohttp_session_init_wrapper"):
+        traces = kwargs.get("trace_configs") or []
+        traces.append(trace_config)
+        kwargs.update({"trace_configs": traces})
+    return func(*args, **kwargs)
+
+
+async def on_request_start(session, trace_config_ctx, params):
+    with lumigo_safe_execute("aiohttp on_request_start"):
+        span = add_request_event(
+            span_id=None,
+            parse_params=HttpRequest(
+                host=params.url.host,
+                method=params.method,
+                uri=str(params.url),
+                headers=dict(params.headers),
+                body=b"",
+            ),
+        )
+        setattr(trace_config_ctx, LUMIGO_SPAN_ID_KEY, span["id"])
+
+
+async def on_request_chunk_sent(session, trace_config_ctx, params):
+    with lumigo_safe_execute("aiohttp on_request_chunk_sent"):
+        span_id = getattr(trace_config_ctx, LUMIGO_SPAN_ID_KEY)
+        span = SpansContainer.get_span().get_span_by_id(span_id)
+        http_info = span.get("info", {}).get("httpInfo", {})
+        http_info["request"]["body"] = concat_old_body_to_new(
+            http_info.get("request", {}).get("body"), params.chunk
+        )
+
+
+async def on_request_end(session, trace_config_ctx, params):
+    with lumigo_safe_execute("aiohttp on_request_end"):
+        span_id = getattr(trace_config_ctx, LUMIGO_SPAN_ID_KEY)
+        update_event_response(
+            span_id, params.url.host, params.response.status, dict(params.response.headers), b""
+        )
+
+
+async def on_response_chunk_received(session, trace_config_ctx, params):
+    with lumigo_safe_execute("aiohttp on_response_chunk_received"):
+        span_id = getattr(trace_config_ctx, LUMIGO_SPAN_ID_KEY)
+        span = SpansContainer.get_span().get_span_by_id(span_id)
+        http_info = span.get("info", {}).get("httpInfo", {})
+        http_info["response"]["body"] = concat_old_body_to_new(
+            http_info.get("response", {}).get("body"), params.chunk
+        )
+
+
+async def on_request_exception(session, trace_config_ctx, params):
+    with lumigo_safe_execute("aiohttp on_request_exception"):
+        span_id = getattr(trace_config_ctx, LUMIGO_SPAN_ID_KEY)
+        span = SpansContainer.get_span().get_span_by_id(span_id)
+        SpansContainer.add_exception_to_span(span, params.exception, [])
+
+
+def wrap_aiohttp():
+    with lumigo_safe_execute("wrap http calls"):
+        get_logger().debug("wrapping http requests")
+        if aiohttp:
+            wrap_function_wrapper(
+                "aiohttp.client", "ClientSession.__init__", aiohttp_session_init_wrapper
+            )
+            global trace_config
```

Comment:
Why use a global variable and set it inside a function?
You can pass it to each function if you don't want to create it inside each function.
Also, I think it is used only in one function.

Reply:
It is indeed used only by `aiohttp_session_init_wrapper`, but it's signature is dictated by `wrap_function_wrapper`.
I can wrap it with another function, but it might be very confusing. WDYT?

Reply:
I think it's better to wrap it with another function.
Global variables might hit us in places we don't expect (warm starts, unittests etc.), better to avoid stateful when possible.